src/crafting.cpp

```diff
@@ -739,29 +739,42 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
     } else { // Let the player pick which component they want to use
         uimenu cmenu;
         // Populate options with the names of the items
+        std::string rot;
```

Comment:
This variable is not used outside of the loops, so there is no need to define it outside of the loops. Change the lines that reset it (`rot = "";`) to a simple definition (`std::string rot;`).